### DISCUSSION TITLE
How to get old validation error message in Laravel 9.x ??

### DISCUSSION THREAD

**Discussion by a-h-abid** [laravel/framework#43130] [Q&A] [answered]
After upgraidng the app from Laravel 6 -> 9, I see that the validation error message in 422 API response has been changed.

**Before:**
```
{
  "message": "The given data was invalid."   <--- This got changed
  "errors": {
    "field1": ["The field1 field is required."],
   ...
  }
}
```

**After:**
```
{
  "message": "The field1 field is required. (and 2 more errors)"   <--- This is new one
  "errors": {
    "field1": ["The field1 field is required."],
   ...
  }
}
```

How can I get the old one back ?

**Comment by erikn69** [comment, marked as answer]
It was changed on https://github.com/laravel/framework/commit/1e02bc2b1508a9a73f76f33405053278983064f0
For get old one back you have to create a class extending `\Illuminate\Validation\ValidationException` and overwrite the new code with the old one
and set that new custom class on validator with `->setException(CustomValidationException::class)`
https://github.com/laravel/framework/blob/c8c541ccaf270cddf5edc710c37708101f154b29/tests/Validation/ValidationValidatorTest.php#L736-L750

**Reply by henzeb** [reply]
I still wonder why one would want that message anyway. Couldn't you just replace it with your own message in your frontend app?

**Reply by a-h-abid** [reply]
> I still wonder why one would want that message anyway. Couldn't you just replace it with your own message in your frontend app?

As our application is large and we don't want to go update every API right now.

BTW @erikn69 , thanks for the suggestion. I created the custom exception class. However I could not find the way to set the custom exception in the validator. I think I have to do it somewhere in AppServiceProvider (or similar), but not sure how.

**Reply by erikn69** [reply]
@henzeb in laravel 6.x you can add any data type as field validation error, but on laravel 9 with error summarize not, it throws error, and if you are upgrading is easy rollback that line that change all the old application

**Reply by erikn69** [reply]
@a-h-abid it depends on your application, my app handles validator globally

**Comment by a-h-abid** [comment]
I actually found later after my post here, but didn't shared then. Thought I should.

So to change the message, add/override below method in the `App\Exceptions\Handler` class:

```
    protected function invalidJson($request, ValidationException $exception)
    {
        return response()->json([
            'message' => "The given data was invalid.",
            'errors' => $exception->errors(),
        ], $exception->status);
    }
```

Also, I recently made a [pull request](https://github.com/laravel/framework/pull/48736) which allows to use custom validation exception class in form requests.

Set to, add this to AppServiceProvider.

```
\Illuminate\Support\Facades\Validator::resolver(function ($translator, $data, $rules, $messages, $customAttributes) {
    $validator = new \Illuminate\Validation\Validator($translator, $data, $rules, $messages, $customAttributes);
    $validator->setException(MyValidationException::class);
    return $validator;
});
```
